utils.groovy

```diff
@@ -282,4 +282,12 @@ def get_artifacts_to_build(pipecfg, stream, basearch) {
     }
     return artifacts.unique()
 }
+
+def get_registry_repos(pipecfg, stream) {
+    def registry_repos = pipecfg.registry_repos ?: [:]
```

Comment:
what does `?:` do?

Reply:
It's like the GCC extension ternary operator (though pretty sure most mainstream C compilers support it too by now): https://gcc.gnu.org/onlinedocs/gcc/Conditionals.html#Conditionals

Reply:
oh wow.. I've only ever used the `x ? x : y` form. I didn't know `x ? : y` existed. How did I not know this?

For future me could we just the longer form?

Reply:
if not could we at least separate the `?:` into `? :` ?

Reply:
Trust me, once you use `?:`, you'll want to keep using it. :) It's a really nice QOL code improvement. The canonical way to type it [in most languages](https://en.wikipedia.org/wiki/Elvis_operator) is `?:` and that's how Groovy [documents it](http://groovy-lang.org/operators.html#_elvis_operator).